### PR TITLE
first ticket 4 all reviews

### PR DESCRIPTION
**Pull request by Daniel17971**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,23 @@
+import { Route, Routes } from "react-router-dom";
 import "./App.css";
+// @ts-ignore
```
Comment (callumu21) [inline src/App.js:3]: Not too sure why we're adding TypeScript syntax here 🤔 

You should go ahead and remove this

File: src/api.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
```
Comment (callumu21) [inline src/api.js:1]: Really nice job here. I would also look in the notes at creating a custom Axios instance to save you having to repeat the base URL of your BE every time

File: src/components/AllReviews.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+const AllReview = () => {
+  const [allReviews, setAllReviews] = useState([]);
+  const [page, setPage] = useState("1");
+  const [next, setNext] = useState("");
+  const [previous, setPrevious] = useState("");
+  useEffect(() => {
+    fetchReviews(page).then((data) => {
+      setAllReviews(data.results);
+      setNext(data.next.page);
+      setPrevious(data.previous.page);
```
Comment (callumu21) [inline src/components/AllReviews.jsx]: On the first page, there is no data.previous that you have access to. You will need to find a way around this, as currently it's throwing an error.

File: src/components/AllReviews.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                <p>Category : {review.category}</p>
+                <p>User : {review.owner}</p>
+                <p>Votes : {review.votes}</p>
+              </li>
+            );
+          })}
+        </ul>
+      </main>
+      <nav>
+        <button onClick={handleClick} value="">
```
Comment (callumu21) [inline src/components/AllReviews.jsx]: Should I be able to press previous when I'm on page 1? Currently, it throw a 400 error

File: src/components/AllReviews.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react";
+import { fetchReviews } from "../api";
+
+const AllReview = () => {
+  const [allReviews, setAllReviews] = useState([]);
+  const [page, setPage] = useState("1");
+  const [next, setNext] = useState("");
+  const [previous, setPrevious] = useState("");
+  useEffect(() => {
```
Comment (callumu21) [inline src/components/AllReviews.jsx:10]: We should have a loading state for our fetch requests to show users that the page is loading whilst the data is being fetched.

**Review by callumu21 — CHANGES_REQUESTED** [review]
Great start, and nice work beginning with pagination. Just a few changes that need to be made:

1. Your components that depend on a fetch request should have a loading state with visual feedback to a user that the page is still loading.
2. You are currently setting a previous page on initial render, but you are currently getting undefined back from your API request. I would add some logic to ensure that previous and next are only set if the relevant data is returned.

Optionally, I would also recommend looking at:

1. Making it so that the previous button can only be pressed if the value of the current page is more than 1. Currently,  a user can press it and Axios throws a 400 error code.
2. Consider rejigging some of your styling, as it looks like your homepage introduction is squashed down the left-hand side of your homescreen. I would place it above the featured reviews list for better readability.

Once you've incorporated the relevant points, please resubmit this PR and we'll take another look 👍

File: src/api.js (hunk 2 above)
Comment (Hard-Luck) [reply src/api.js:1]: I would second this! ^^ look at axios.create

File: src/App.css [hunk 6]
```diff
@@ -1,38 +1,96 @@
...
+}
+/* for the Header and nav bar */
+header {
+  grid-template-columns: auto auto;
+  grid-template-rows: auto;
+  display: grid;
+  padding: 2rem;
+  background-color: #810bc6;
+  margin-bottom: 1rem;
+  border-radius: 5%;
```
Comment (Hard-Luck) [inline src/App.css]: This is minor but.. I would generally use an absolute value for border radius, as then you have predictability of roundness regardless of screen size.

File: src/components/AllReviews.jsx (hunk 4 above)
Comment (Hard-Luck) [reply src/components/AllReviews.jsx]: This still hasn't been implemented.

**Review by Hard-Luck — COMMENTED** [review]
The button rendering conditionally is still something to be addressed and I would highly recommend creating the axios instance. The note about border radius is something I would also take into account, though this is just stylistic.

**Pull request merged**